app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -126,6 +147,16 @@ def network_client
       @network_client = nil
     end
 
+    def volume_client
+      @volume_client ||= ::Fog::Volume.new(:provider           => :openstack,
+                                             :openstack_api_key  => password,
+                                             :openstack_username => user,
+                                             :openstack_auth_url => url,
+                                             :openstack_tenant   => tenant)
+    rescue
```

Comment:
The thought was in case of some error connecting to Openstack API, but if you think this is not necessary I can remove.